tools/lib/provision_inner.py

```diff
@@ -80,9 +80,9 @@ def main(options: argparse.Namespace) -> int:
     # The `build_emoji` script requires `emoji-datasource` package
     # which we install via npm; thus this step is after installing npm
     # packages.
-    if not os.path.isdir(EMOJI_CACHE_PATH):
-        run_as_root(["mkdir", EMOJI_CACHE_PATH])
-    run_as_root(["chown", "%s:%s" % (user_id, user_id), EMOJI_CACHE_PATH])
+    if not os.access(EMOJI_CACHE_PATH, os.W_OK):
+        run_as_root(["mkdir", "-p", EMOJI_CACHE_PATH])
+        run_as_root(["chown", "%s:%s" % (user_id, user_id), EMOJI_CACHE_PATH])
```

Comment:
This seems not equivalent, in that we'll now attempt to create it if it's owned by someone else and fail?

Reply:
`mkdir -p` succeeds if it already exists and is owned by someone else.

Reply:
Is that true even if you don't have read access to the directory?

Reply:
I guess probably the bug here was long before the Bionic upgrade in any case.

Reply:
> Is that true even if you don't have read access to the directory?

Yes, but in any case the `mkdir` runs as `root`, so I don’t understand the concern about not having read access.